File: Java/1957.java
class Solution {
    public String makeFancyString(String s) {
        if (s.length() < 3) {
            return s;
        }
        StringBuilder result = new StringBuilder();
        char rear = s.charAt(0), middle = s.charAt(1);
        result.append(rear);
        result.append(middle);
        for (int i = 2; i < s.length(); i++) {
            char current = s.charAt(i);
            if (s.charAt(i) == middle && middle == rear) {
                continue;
            }
            rear = middle;
            middle = current;
            result.append(current);
        }
        return result.toString();
    }
}

File: Python/1957.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        if len(s) < 3:
            return s

        s_list = list(s)
        j = 2

        for i in range(2, len(s)):
            if s_list[i] != s_list[j - 1] or s_list[i] != s_list[j - 2]:
                s_list[j] = s_list[i]
                j += 1

        return "".join(s_list[:j])
